File: Go 19.08.2022/Source.go
/*
  The source code is authored by the following CodeWars User: www.codewars.com/users/akar-0
*/

package kata

import (
  "math"
  "fmt"
)

func FindTheNumberPlate(n int) string {
  var res []rune
  for _,i := range []float64{2.0,1.0,0.0} {
    d := 999 * int(math.Pow(26.0, i))
    l := n / d
    n %= d
    res = append([]rune{rune(l+97)}, res...)
  }
  return string(res) + fmt.Sprintf("%03d", n+1)
}

File: C 26.08.2022/Source.c
/*
  The source code was authored by the following CodeWars user: www.codewars.com/users/porkfreezer
*/

#include <stdlib.h>
#include <stdio.h>

char *find_the_number_plate(int id) {
  char *out = calloc(7, 1);
  int sn = id % 999 + 1;
  char l1 = (id /= 999) % 26 + 'a';
  char l2 = (id /= 26) % 26 + 'a';
  char l3 = (id / 26) % 26 + 'a';
  sprintf(out, "%c%c%c%03d", l1, l2, l3, sn);
  return out;
}
